doc/clients.md

```diff
@@ -0,0 +1,96 @@
+# Jackdaw Client API
```

Comment:
Should the filename be `doc/client.md` without the 's'?

Reply:
The thinking behind the plural was that this covered both types of clients, producers and consumers. I can appreciate the argument for making it match the final segment of the namespace. Not too fussed either way.

Reply:
Two comments:
1. Kafka treats the consumer and producer as separate APIs, but in Jackdaw we treat these as a single "Client API". 
2. I was thinking we may want to match the Clojure namespace the documentation is for.

This is not a blocker.

Reply:
Agreed. I've updated to client.md